### PR TITLE
[fix][AB-1210] not working after several hours

### PR DESCRIPTION
**Pull request by oShcherbininSuper**

### CODE REVIEW

**Review by httpdispatch — APPROVED** [review]

File: src/frontends/android/app/src/main/jni/libandroidbridge/backend/android_service.c [hunk 1]
```diff
@@ -157,6 +157,7 @@ static job_requeue_t handle_plain(private_android_service_t *this)
     struct pollfd pfd_read;
     int timeout = 1000;
     pfd_read.fd = tunfd;
+    pfd_read.events = POLLIN;
```
Comment (httpdispatch) [inline src/frontends/android/app/src/main/jni/libandroidbridge/backend/android_service.c:160]: - Does it work?
- Only POLLIN is required? How about POLLLOUT?

Comment (oShcherbininSuper) [reply src/frontends/android/app/src/main/jni/libandroidbridge/backend/android_service.c:160]: works with POLLIN

File: src/frontends/android/app/src/main/jni/Android.mk [hunk 2]
```diff
@@ -1,4 +1,7 @@
 LOCAL_PATH := $(call my-dir)
+LOCAL_CFLAGS := -g
+LOCAL_STRIP_MODE := none
+LOCAL_STRIP_MODULE := keep_symbols
```
Comment (httpdispatch) [inline src/frontends/android/app/src/main/jni/Android.mk:4]: Could you please describe why it is required?

Comment (oShcherbininSuper) [reply src/frontends/android/app/src/main/jni/Android.mk:4]: To generate debug symbols

Comment (oShcherbininSuper) [reply src/frontends/android/app/src/main/jni/Android.mk:4]: To keep them in generated libs

File: src/frontends/android/app/src/main/jni/libandroidbridge/backend/android_service.c (hunk 1 above)
Comment (oShcherbininSuper) [reply src/frontends/android/app/src/main/jni/libandroidbridge/backend/android_service.c:155]: It is only to read info, the same logic was with select

File: src/frontends/android/app/src/main/jni/libandroidbridge/backend/android_service.c [hunk 3]
```diff
@@ -135,45 +135,51 @@ static job_requeue_t handle_plain(private_android_service_t *this)
...
+        }
+        DBG1(DBG_DMN, "poll on TUN device failed: %s", strerror(errno));
+        return JOB_REQUEUE_NONE;
 	}
 	else if (len == 0)
 	{	/* timeout, check again right away */
-		return JOB_REQUEUE_FAIR;
-	}
+        DBG1(DBG_DMN, "poll file descriptor timeout");
+        return JOB_RESCHEDULE_MS(3001);
```
Comment (oShcherbininSuper) [inline src/frontends/android/app/src/main/jni/libandroidbridge/backend/android_service.c:171]: > 3000 will run external scheduler

File: src/frontends/android/app/src/main/jni/libandroidbridge/backend/android_service.c [hunk 4]
```diff
@@ -135,45 +135,51 @@ static job_requeue_t handle_plain(private_android_service_t *this)
...
-		DBG1(DBG_DMN, "select on TUN device failed: %s", strerror(errno));
-		return JOB_REQUEUE_NONE;
+        if (errno == EBADF)
+        {	/* the TUN device got closed just before calling poll(), retry */
+            return JOB_REQUEUE_FAIR;
+        }
+        DBG1(DBG_DMN, "poll on TUN device failed: %s", strerror(errno));
+        return JOB_REQUEUE_NONE;
 	}
 	else if (len == 0)
```
Comment (oShcherbininSuper) [inline src/frontends/android/app/src/main/jni/libandroidbridge/backend/android_service.c:168]: It means the system will not send any network requests by this tymeout

**Pull request closed without merging**